### PR TITLE
Remove special case for stdlib in IsNonUserModuleRequest

### PR DESCRIPTION
**Pull request by hamishknight**
It's possible this may be the main module, or a "local" dependency within the same project, and should be treated as a user module in such cases.

### CODE REVIEW

**Comment by hamishknight** [comment]
@swift-ci please test

**Comment by hamishknight** [comment]
@swift-ci please build toolchain

File: test/SourceKit/CursorInfo/Inputs/Swift.swift [hunk 1]
```diff
@@ -0,0 +1,4 @@
+// This is built with '-parse-stdlib', so we can't use any stdlib types.
```
Comment (bnbarham) [inline test/SourceKit/CursorInfo/Inputs/Swift.swift:1]: Could you use `split-file` instead 🙏 I find it makes these so much easier to follow outside of a review (ie. when trying to work out what a test does after the fact).

**Review by bnbarham — APPROVED** [review]

**Review by xymus — APPROVED** [review]

**Comment by hamishknight** [comment]
@swift-ci please smoke test

**Comment by hamishknight** [comment]
@swift-ci please smoke test

**Pull request merged**